tls_icon.php

```diff
@@ -72,7 +74,7 @@ public function message_headers($p)
 		}
 
 		if (isset($p['output']['subject'])) {
-			$p['output']['subject']['value'] = htmlentities($p['output']['subject']['value']) . $this->icon_img;
```

Comment:
The `htmlentities` is there for a very good reason: It prevents HTML & JavaScript from executing on subjects. Without this, any HTML that is already there from the start might be interpreted which opens the door to XSS security vulnerabilities.

I know that this may bite with other plugins that also modify the subject line, but removing this is **not** the solution. If no other plugin sanitizes the user input, we have a glaring XSS security problem here.

If we wanted to make this optional, we would at least need a default-off "unsafe disable subject sanitization"-config to allow users to decide whether they already have a plugin that filters incoming subject lines. Alternatively, if there is a way to reliably detect whether subjects have been sanitized by another plugin (perhaps we could look at the subject-html key below? Is that reliable?) we could auto-decide whether sanitization is required or not.

In any case, this sounds like something that needs more thought and is probably best on a separate PR.

Reply:
I understand your idea and I missed that in this design. I think, a check for what the receiving server is in the regex hit, is a good solution.

Reply:
I agree. Probably a config option to set servername(s) we should be looking for and then find them with a regex.